packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1256,6 +1287,7 @@ export async function renderToHTMLOrFlight(
               notFound={
                 NotFound && RootLayout ? (
                   <RootLayout params={{}}>
+                    {assets}
```

Comment:
This feels off - they’re not not-found specific assets and they should be on the layout level:

```
{assets}
<RootLayout>
  {notFoundStyles}
  <NotFound/>
</RootLayout>
```

If it’s due to some other reason like how we structured the layouts with special logic for not found, I feel that something is wrong in our structure and there could be other bugs.